src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
```

Comment:
This class has several separate responsibilities: handling the basket, basket price calculation, local product handling, and promo code application. By applying the Single Responsibility Principe, we can separate the above class into multiple classes with separate responsibilities.